### PR TITLE
test: add test coverage for EditFileTool and patch subsystem

### PR DESCRIPTION
**Pull request by BLannoo**
## Summary

Adds comprehensive test coverage for `EditFileTool` and its underlying patch subsystem (`FilePatch`, `TokenNormalizedPatchApplier`), which previously had **zero test coverage**.

### `EditFileToolJvmTest.kt` (16 tests)
JVM integration tests for the `EditFileTool`, following the same patterns as existing sibling tests (`WriteFileToolJvmTest`, `ReadFileToolJvmTest`).

Covers:
- **Descriptor validation** — tool name, description, required/optional parameters
- **Args data class** — field mapping
- **Text replacement** — basic replace in existing file
- **Rewrite mode** — creating new files with empty original
- **Parent directory creation** — nested path handling for new files
- **Text deletion** — empty replacement removes matched content
- **File rewrite** — empty original on existing file replaces all content
- **Failure case** — original text not found returns failure, file unchanged
- **Binary file validation** — throws `ValidationFailure` for non-text files
- **Case-insensitive matching** — fuzzy case matching via token normalization
- **Whitespace-insensitive matching** — fuzzy whitespace matching
- **Multiline replacement** — cross-line edit operations
- **Content preservation** — surrounding content unaffected by patch
- **Result rendering** — `textForLLM()` output for success and failure
- **No-op edge case** — both original and replacement empty
- **First-occurrence-only** — replaces only the first match

### `FilePatchTest.kt` (5 tests)
Common (multiplatform) unit tests for the `FilePatch` data class.

Covers:
- `isDelete` — non-empty original + empty replacement
- `isReplace` — both non-empty and different
- `isRewrite` — empty original + non-empty replacement
- Identical original/replacement — no flags set
- Both empty — no flags set

### `TokenNormalizedPatchApplierTest.kt` (20 tests)
Common (multiplatform) unit tests for the tokenizer, `TokenList`, and `applyTokenNormalizedPatch`.

Covers:
- **Tokenizer** — whitespace splitting, newline splitting, punctuation splitting, empty input, range correctness, multi-separator types, whitespace/non-whitespace token identification
- **TokenList.find** — matching subsequence, no match, custom equality, first-occurrence semantics
- **TokenList.replace** — mid-list, start, and end replacements
- **applyTokenNormalizedPatch** — replace, delete, rewrite, no-op, case-insensitive, whitespace-insensitive, tab-vs-space, multiline, first-occurrence-only, `OriginalNotFound` failure
- **PatchApplyResult.isSuccess** — success and failure variants

## Motivation

`EditFileTool.kt` and its patch dependencies (`FilePatch`, `TokenNormalizedPatchApplier`) had no automated tests. All sibling file tools (`ReadFileTool`, `WriteFileTool`, `ListDirectoryTool`) already have thorough test suites. This PR closes that gap.

### CODE REVIEW

File: agents/agents-ext/src/jvmTest/kotlin/ai/koog/agents/ext/tool/file/EditFileToolJvmTest.kt [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+    @Test
+    fun `no-op when both original and replacement are empty`() = runBlocking {
+        val f = createTestFile("noop.txt", "content stays")
+
+        val result = edit(f, "", "")
+
+        // Empty original + empty replacement: patch is neither delete, replace, rewrite
+        // so the content stays unchanged, but it returns Success with original content
+        assertTrue(result.applied)
```
Comment (BLannoo) [inline agents/agents-ext/src/jvmTest/kotlin/ai/koog/agents/ext/tool/file/EditFileToolJvmTest.kt:206]: Nit: These substring checks are a bit fragile — `contains("not")` would match unrelated words like "nothing" or "note". Consider matching a more specific substring (e.g., `"not modified"` or the full expected message) so the test doesn't pass by accident if the wording changes.

```kotlin
assertTrue(result.textForLLM().contains("not modified"))
```

Not blocking — the intent is clear and the test works correctly today.

**Review by BLannoo — COMMENTED** [review]
Thorough, well-structured test suite — nice work closing this coverage gap. **Verdict: Approve** (posted as comment since GitHub blocks self-approval).

**What I checked:**
- All 3 source files under test (`EditFileTool`, `FilePatch`, `TokenNormalizedPatchApplier`) — tests are correct against actual implementation behavior.
- Internal API visibility: `commonTest` files properly access `internal` symbols without annotation; `EditFileToolJvmTest` correctly uses `@OptIn(InternalAgentToolsApi::class)` for `DirectToolCallsEnabler`.
- Follows existing sibling test patterns (`WriteFileToolJvmTest`, `ReadFileToolJvmTest`).
- CI failure on macOS is an unrelated flaky port-availability test in `KoogToolAsMcpToolTest`.

One minor nit on a fragile `textForLLM` assertion — otherwise clean.
